### PR TITLE
Add Sentry error ID middleware

### PR DESCRIPTION
**Pull request by shalvah-gs**
Part of [IPDE-813](https://hellogetsafe.atlassian.net/browse/IPDE-813)

Adds a middleware that will include the ID of any error reported to Sentry in 5xx responses.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# v0.4.0 2022-11-14
+- Add `SentryErrorID` middleware
```
Comment (AkihikoITOH) [inline CHANGELOG.md:2]: Also "Drop support for Ruby 2.x"?

File: .circleci/config.yml [hunk 2]
```diff
@@ -10,7 +10,9 @@ workflows:
       - build:
           matrix:
             parameters:
-              ruby-version: ["2.6.8", "2.7.4", "3.0.2"]
+              ruby-version: ["3.0.2", "3.1.2"]
```
Comment (AkihikoITOH) [inline .circleci/config.yml:13]: All our services have already migrated to Ruby 3, right?

File: lib/nxt_support/middleware/sentry_error_id.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    class SentryErrorID
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        status, headers, body = @app.call(env)
+        if status >= 500
+          headers["Sentry-Error-ID"] = env['sentry.error_event_id'] if env['sentry.error_event_id']
+        end
```
Comment (vrudnev) [inline lib/nxt_support/middleware/sentry_error_id.rb]: I think we can combine conditions here, looks more readable. What do you think?
```suggestion
        if status >= 500 && env['sentry.error_event_id']
          headers["Sentry-Error-ID"] = env['sentry.error_event_id']
        end
```

**Review by vrudnev — APPROVED** [review]

**Review by AkihikoITOH — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (AkihikoITOH) [reply .circleci/config.yml:13]: And maybe add `spec.required_ruby_version = '>= 3.0.0'` to gemspec?

File: lib/nxt_support/middleware/sentry_error_id.rb (hunk 3 above)
Comment (shalvah-gs) [reply lib/nxt_support/middleware/sentry_error_id.rb]: True

File: CHANGELOG.md (hunk 1 above)
Comment (shalvah-gs) [reply CHANGELOG.md:2]: 👍

File: .circleci/config.yml (hunk 2 above)
Comment (shalvah-gs) [reply .circleci/config.yml:13]: :+1:

**Pull request merged**